perl/meson_options.txt

```diff
@@ -0,0 +1,50 @@
+# Nix-Perl build options
+#============================================================================
+
+
+# compiler args
+#============================================================================
+
+option(
+    'version',
+    type : 'string',
+    description : 'nix-perl version')
```

Comment:
This is an unfortunate requirement, since it means that outside of configuration through the nix derivation the version will be unset. Meson can actually read a file formatted like the `.version` file is, and automatically set the version by adding `version: files('.version')` to the `project()` call, if you can symlink the `.version` file into the nix-perl directory

Reply:
it would be good to have a meson builtin something like `meson setup build -Dmeson.project_version="1.2.3"`

Reply:
That doesn't seem useful to me. A project version is what it is, having a user arbitrarily change is asking for trouble.

I don't see what's wrong with a simple:
```console
cd perl
ln -s .version ../.version
git add .version
```
and adding `version : files('.version')` to the Meson.buidl

Reply:
symlinks aren't portable and can cause problems in different environments. the symlink would need to be created at buildtime. its messy, but it works just fine for now.

Reply:
Then use:
```meson
version: run_command('cat_versionfile.pl', '../version's, check: true).stdout().strip()
```

Or simply

```meson
version: files('../version')
```

Which I believe should work since meson doesn't care where the file is, just that it exists. If it's outside of the build directory and the project is refactored so the version file is not at the right relative location, that's a project issue, not a sandbox bug.

I really think you're overcomplicating this and there's no justification for meson to add a built-in option to erroneously ignore the upstream version number and break ABI for any project that actually uses the version number for something other than printing the meson summary().